assets/src/hooks/usePersistedStateReducer.ts

```diff
@@ -48,32 +48,31 @@ const usePersistedStateReducer = (): [State, Dispatch] => {
 
   const { routeTabsToPush, routeTabsPushInProgress } = state
 
-  const [routeTabsPushRetriesLeft, setRouteTabsPushRetriesLeft] =
-    useState(routeTabsPushRetries)
+  const routeTabsPushRetriesLeft = useRef(routeTabsPushRetries)
```

Comment:
**question:** I'm wondering if it would even be possible to simply use a local variable in `usePersistedStateReducer` rather than a `ref`, though I haven't thought it all the way through yet and my naive attempt at doing it locally isn't working. This logic is really hard to reason about and I'm hoping there's some way to simplify it a bit further but it might not be possible while fully obeying all of the rules of hooks and everything.